lib/manager/components/deployment/DeploymentTableRow.js

```diff
@@ -0,0 +1,197 @@
+}
+
+function dangerIfTruthy (val) {
+  return addClassIfFalsy(!val, 'danger')
+}
+
+type RowProps = {
+  deployment: Deployment,
+  feedSource: Feed,
+  project: Project,
+  version: SummarizedFeedVersion
+}
+
+type ConnectedRowProps = RowProps & {
+  deleteFeedVersion: typeof deploymentActions.deleteFeedVersion,
+  feedVersionPinned: boolean,
+  isVersionEditable: boolean,
+  togglePinFeedVersion: typeof deploymentActions.togglePinFeedVersion,
+  updateVersionForFeedSource: typeof deploymentActions.updateVersionForFeedSource,
+  userCanEditDeployment: boolean,
+  userCanRemoveVersion: boolean
+}
+
+class DeploymentTableRow extends Component<ConnectedRowProps> {
+  _numberToOption = n => ({value: n, label: n})
+
+  _onChangeVersion = option => {
+    const {
+      deployment,
+      feedSource,
+      updateVersionForFeedSource
+    } = this.props
+    const versionToAdd = {feedSourceId: feedSource.id, version: option.value}
+    updateVersionForFeedSource(deployment, feedSource, versionToAdd)
+  }
+
+  _togglePinFeedVersion = () => {
+    const {togglePinFeedVersion, deployment, version} = this.props
+    togglePinFeedVersion(deployment, version)
+  }
+
+  _removeVersion = () => {
+    const {deleteFeedVersion, deployment, version} = this.props
+    deleteFeedVersion(deployment, version)
+  }
+
+  render () {
+    const {
+      feedSource,
+      feedVersionPinned,
+      isVersionEditable,
+      userCanEditDeployment,
+      userCanRemoveVersion,
+      version
+    } = this.props
+    const {latestVersionId} = feedSource
+    const {validationResult: result} = version
+    const expired = versionHasExpired(version)
+    const future = versionHasNotBegun(version)
+    return (
+      <tr>
+        <td>
+          <Link to={`/feed/${feedSource.id}`}>{feedSource.name}</Link>
+          {feedVersionPinned && <Icon type='thumb-tack' />}
+        </td>
+        <td className={version.id !== latestVersionId ? 'warning' : ''}>
+          <div style={{display: 'flex'}}>
+            <Button
+              bsSize='xsmall'
+              active={feedVersionPinned}
+              disabled={!userCanEditDeployment}
+              onClick={this._togglePinFeedVersion}>
+              <Icon
+                type={'thumb-tack'}
+                title={feedVersionPinned
+                  ? 'Unpin version in auto-deployments'
+                  : 'Pin version in auto-deployments'
+                }
+                alt='Toggle pin version' />
+            </Button>
+            <div style={{marginLeft: '5px'}}>
+              <Select
+                clearable={false}
+                disabled={!isVersionEditable}
+                options={Array.from(Array(feedSource.versions).keys())
+                  .reverse()
+                  .map(n => this._numberToOption(n + 1))
+                }
+                value={this._numberToOption(version.version)}
+                onChange={this._onChangeVersion}
+              />
```

Comment:
Any possibility of using `VersionSelectorDropdown`?

Reply:
No, we only have one version available in this case. `FeedSource#versions` is a number field, not a list of versions in this case.

Reply:
This needs to be wider to properly show version numbers above 99 and potentially those above 999 also.